h/h_api/exceptions.py

```diff
@@ -0,0 +1,67 @@
+from h.h_api.model.json_api import JSONAPIError, JSONAPIErrorBody
+
+
+class JSONAPIException(Exception):
```

Comment:
This will form the basis for all our exceptions, so they can be trivial written out as JSON

Reply:
Does this exception base class form part of the lib's public interface? E.g. might lms or h want to catch `JSONAPIException` or subclasses thereof, or have exception views for them?

The `exceptions.py` filename suggests that it is.

If it's part of the public interface then I wonder about calling it `HAPIException` or something like that. I think it's typical for the base class of all exceptions raised by a lib to be named after that lib (`RequestException` in `requests` etc). And I think the fact that JSON API is being used is mostly an internal implementation detail of the `h_api` lib, which lms and h don't need to know about.

Reply:
I also wonder whether `JSONAPIError` / `HAPIError` would be a better name, rather than exception? It looks to me like this class and all its subclasses specifically represent errors

Reply:
I tend to swither about between the two in Python, but my understanding is that they are more properly `Exceptions` rather than `Errors` because errors are something you shouldn't every expect to happen or bother trying to catch. Like a syntax error or something.

An exception is something you do expect to happen during normal execution. How do you define the difference?

Reply:
With regards to interface, yes any caller of this would have to catch them in some way, so they need to be accessible outside. 

Whether you'd want to let them bubble all the way to an error view or catch and transform them into pyramid errors inside the view is TBD.